modules/EPMC.py

```diff
@@ -4,160 +4,171 @@
-        .set('spark.executor.memory', f'{spark_mem_limit}g')
-        .set('spark.driver.maxResultSize', '0')
-        .set('spark.debug.maxToStringFields', '2000')
-        .set('spark.sql.execution.arrow.maxRecordsPerBatch', '500000')
-    )
-    spark = (
-        SparkSession.builder
-        .config(conf=spark_conf)
-        .master('local[*]')
-        .getOrCreate()
-    )
-
-    logging.info(f'Spark version: {spark.version}')
+SECTIONS_OF_INTEREST = [
+    "title",
+    "abstract",
+    "intro",
+    "case",
+    "figure",
+    "table",
+    "discuss",
+    "concl",
+    "results",
+    "appendix",
+    "other",
+]
+
+
+def main(cooccurrences, outputFile):
 
     # Log parameters:
-    logging.info(f'Cooccurrence file: {cooccurrenceFile}')
+    logging.info(f'Cooccurrence file: {cooccurrences}')
     logging.info(f'Output file: {outputFile}')
     logging.info('Generating evidence:')
 
     # Load/filter datasets:
-    filtered_cooccurrence_df = (
+    agg_cooccurrence_df = (
         # Reading file:
-        spark.read.parquet(cooccurrenceFile)
-
+        read_path(cooccurrences, spark)
+        .repartition(200)
         # Filter out pairs found in unwanted sections
-        .filter(pf.col('section').isin(SECTIONS_OF_INTEREST))
-
+        .filter(F.col('section').isin(SECTIONS_OF_INTEREST))
         # Casting integer pmid column to string:
-        .withColumn("pmid", pf.trim(pf.col('pmid').cast(StringType())))
-
+        .withColumn("pmid", F.trim(F.col('pmid').cast(StringType())))
         # Publication identifier is a pmid if available, otherwise pmcid
-        .withColumn(
-            'publicationIdentifier',
-            pf.when(pf.col('pmid').isNull(), pf.col('pmcid'))
-            .otherwise(pf.col('pmid'))
-        )
-
+        .withColumn('publicationIdentifier', F.when(F.col('pmid').isNull(), F.col('pmcid')).otherwise(F.col('pmid')))
         # Filtering for disease/target cooccurrences:
         .filter(
-            (pf.col('type') == 'GP-DS') &  # Filter gene/protein - disease cooccurrence
-            (pf.col('isMapped')) &  # Filtering for mapped cooccurrences
-            (pf.col('publicationIdentifier').isNotNull()) &  # Making sure at least the pmid or the pmcid is given:
-            (pf.length(pf.col('text')) < 600) &  # Exclude sentences with more than 600 characters
-            (pf.col('label1').isin(EXCLUDED_TARGET_TERMS) == False)  # Excluding target labels from the exclusion list
+            (F.col('type') == 'GP-DS')
+            & (F.col('isMapped'))  # Filter gene/protein - disease cooccurrence
+            & (F.col('publicationIdentifier').isNotNull())  # Filtering for mapped cooccurrences
+            & (F.length(F.col('text')) < 600)  # Making sure at least the pmid or the pmcid is given:
+            & (  # Exclude sentences with more than 600 characters
+                F.col('label1').isin(EXCLUDED_TARGET_TERMS) == False
+            )  # Excluding target labels from the exclusion list
```

Comment:
Comments got shifted. Also I have a feeling the brackets are excessively used here:
```suggestion
            (F.col('type') == 'GP-DS')  # Filter gene/protein - disease cooccurrence
            & F.col('isMapped')  # Filtering for mapped cooccurrences
            & F.col('publicationIdentifier').isNotNull(). # Making sure at least the pmid or the pmcid is given:
            & (F.length(F.col('text')) < 600) # Exclude sentences with more than 600 characters
            & (F.col('label1').isin(EXCLUDED_TARGET_TERMS) == False)  # Excluding target labels from the exclusion list
```